server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,448 @@
+    backup_tar = os.path.join(backup_controller_path, resultname)
+    if os.path.exists(backup_tar) and os.path.isfile(backup_tar):
+        backup_md5 = os.path.join(
+            backup_controller_path, "{}.md5".format(resultname))
+
+        # check backup md5 file exist and it is a regular file
+        if os.path.exists(backup_md5) and os.path.isfile(backup_md5):
+            pass
+        else:
+            # backup md5 file does not exist or it is not a regular file
+            logger.error(
+                "{} does not exist or it is not a regular file\n".format(backup_md5))
+            return Status.FAIL
+
+        # read backup md5 file
+        try:
+            with open(backup_md5) as f:
+                backup_md5_hex_value = f.readline().split(" ")[0]
+        except (OSError, IOError) as e:
+            # Could not read file
+            logger.error(
+                "Could not read file {}, {}\n".format(backup_md5, e))
+            return Status.FAIL
+        else:
+            if archive_md5_hex_value == backup_md5_hex_value:
+                # declare success
+                logger.info(
+                    "Already locally backed-up: {}\n".format(resultname))
+                return Status.SUCCESS
+            else:
+                # md5 file of archive and backup does not match
+                logger.error(
+                    "{} already exists in backup but md5 sums of archive and backup disagree\n".format(resultname))
+                return Status.FAIL
+    else:
+        tar_done = False
+
+        # copy the md5 file from archive to backup
+        try:
+            shutil.copy(archive_md5, backup_controller_path)
+        except Exception:
+            # couldn't copy md5 file
+            md5_done = False
+            logger.exception(
+                "shutil.copy: Unable to copy {} from archive to backup: {}\n".format(archive_md5, backup_controller_path))
+        else:
+            md5_done = True
+
+        # copy the tarball from archive to backup
+        if md5_done:
+            try:
+                shutil.copy(tar, backup_controller_path)
+            except Exception:
+                # couldn't copy tarball
+                tar_done = False
+                logger.exception(
+                    "shutil.copy: Unable to copy {} from archive to backup: {}\n".format(tar, backup_controller_path))
+
+                # remove the copied md5 file from backup
+                bmd5_file = os.path.join(
+                    backup_controller_path, "{}.md5".format(resultname))
+                if os.path.exists(bmd5_file):
+                    try:
+                        os.remove(bmd5_file)
+                    except Exception:
+                        logger.exception("Unable to remove: {}".format(bmd5_file))
+            else:
+                tar_done = True
+
+        if md5_done and tar_done:
+            logger.info(
+                "Locally Backed-up Sucessfully: {}\n".format(resultname))
```

Comment:
I think all the log messages that report `resultname` should also report `controller`: the resultname by itself is probably unambiguous, but finding it in the archive is made much easier if we know the controller as well.